src/dataflow/components/stats.tsx

```diff
@@ -0,0 +1,276 @@
+import { inject, observer } from "mobx-react";
+import { BaseComponent, IBaseProps } from "./dataflow-base";
+import * as React from "react";
+import { exportCSV } from "../utilities/export";
+import "./stats.sass";
+
+export interface StatsGroup {
+  studentName: string;
+  documents: number;
+  programDocuments: number;
+  runProgramDocuments: number;
+  runProgramDataDocuments: number;
+  runProgramRelayOnlyDocuments: number;
+  publishedProgramDocuments: number;
+}
+
+interface IProps extends IBaseProps {}
+
+interface IState {
+  loading: boolean;
+  userStats: StatsGroup[];
+}
+
+const kHeaderNames = ["Documents", "Editable Programs", "Run Programs",
+                      "Run Programs w/ Data", "Run Programs Relay-Only", "Published Programs"];
+
+@inject("stores")
+@observer
+export class StatsComponent extends BaseComponent<IProps, IState> {
+
+  constructor(props: IProps) {
+    super(props);
+    this.state = {
+      loading: true,
+      userStats: [],
+    };
+  }
+
+  public async componentDidMount() {
+    const {db} = this.stores;
+    // request the published documents
+    const publicationsPath = db.firebase.getClassPersonalPublicationsPath(db.stores.user);
+    const publicationsRef = db.firebase.ref(publicationsPath);
+    const publicationsSnapshot = await publicationsRef.once("value");
+    const publishedDocOriginIds: string[] = [];
+    if (publicationsSnapshot) {
+      const pulishedDocs = publicationsSnapshot.val();
+      Object.keys(pulishedDocs).forEach((pulishedDoc) => {
```

Comment:
```suggestion
      each(publishedDocs, publishedDoc => {
```
Not necessarily a suggestion, as some would argue that native code should be preferred over a Lodash library function, but in this case I find the Lodash usage leads to code that's easier to read.

Reply:
I usually prefer the native code, but the Lodash `each` is so much slimmer, and, I agree, easier to read.  The `Object.keys` is clunky looking.  I went with your Lodash `each` suggestion.